projects/plugins/jetpack/tests/php/sync/test_class.jetpack-sync-modules-stats.php

```diff
@@ -29,14 +33,21 @@ function test_sends_stats_data_on_heartbeat() {
 	 * @expectedDeprecated Jetpack_Heartbeat::cron_exec
 	 * @return void
 	 */
-	function test_dont_send_expensive_data_on_heartbeat() {
+	public function test_dont_send_expensive_data_on_heartbeat() {
+		$this->make_site_active();
+
 		$heartbeat = Jetpack_Heartbeat::init();
+		add_filter( 'jetpack_heartbeat_stats_array', array( $heartbeat, 'add_stats_to_heartbeat' ) );
```

Comment:
@kbrown9 Kim do you mind highlighting why this filter needs to be added for the tests?  I'm looking at the Jetpack_Heartbeat class vs the Heartbeat package and want to make sure the test is aligned

Reply:
With the introduction of the Heartbeat package, Jetpack's heartbeat stats are added to the generated stats array via the filter. [This PR](https://github.com/Automattic/jetpack/pull/16285) is where that change was made. If you're wondering why the filter has to be set in the test when it was [already set](https://github.com/Automattic/jetpack/blob/master/projects/plugins/jetpack/class.jetpack-heartbeat.php#L47) in the `Jetpack_Heartbeat` constructor, that's necessary because the filters are reset between tests.

I wanted to maintain the original intention of the tests (or at least what I interpreted as the original intention), which is to verify that Jetpack's heartbeat stats make it to Sync.

I made some changes to the tests in commit 9dc9de7:
 * Added new tests that do not test the integration with Jetpack. Instead, a test stat is added via the `jetpack_heartbeat_stats_array` filter in the test class. This is closer to a unit test than the original tests (but they still test the integration with the heartbeat package).
  * Added comments in the `..._jetpack_integration` tests, which indicate that the tests integrate with Jetpack, e.g. [this](https://github.com/Automattic/jetpack/pull/18494/files#diff-fe03236b34513c0c6387c69249b1e9957ebc7adc6c2cf7d40a2770cecdc843f7R37) and [this.](https://github.com/Automattic/jetpack/pull/18494/files#diff-fe03236b34513c0c6387c69249b1e9957ebc7adc6c2cf7d40a2770cecdc843f7R47)
  
Do you think the additional comments in the test methods provide enough information?

Reply:
@mdbitz Are you good with Kim's reply? I'm approving the PR but would appreciate confirmation that you're happy. :)

Reply:
Definitely, sorry this totally slipped my radar.